xpdan/tools.py

```diff
@@ -240,3 +256,53 @@ def decompress_mask(data, indices, indptr, shape):
     cmask = csr_matrix(
         tuple([np.asarray(a) for a in [data, indices, indptr]]), shape=shape)
     return ~cmask.toarray().astype(bool)
+
+
+def auto_polarization(img, npt, geo, mask=None):
+    """Optimize the polarization by minimizing the azimuthal standard
+    deviation
+
+    Parameters
+    ----------
+    img: np.ndarray
+        The image
+    npt: int
+        The number of points used for integration
+    geo: pyFAI.geometry.Geometery instance or subclass
+        The pyFAI geometry class or subclass obtained from calibrating the
+        detector
+    mask: np.ndarray bool, optional
+        The boolian array which represents the mask, if none then build a mask
+        internally
+
+    Returns
+    -------
+    float:
+        The polarization factor
+    """
+    q = geo.qArray(img.shape)
+    q_range = np.linspace(np.min(q), np.max(q), npt)
+
+    pol = None
+    img3 = img.copy()
+    if pol:
+        img3 /= geo.polarization(img.shape, .95)
+    if not mask:
+        mask = mask_img(img3, geo, alpha=3, bs_width=None)
+
+    def opt(polarization_factor):
```

Comment:
it would be nice to do more testing on this.

Some thoughts (some of these things should be captured as Issues for a future release):
1) do we want a function that can find Bragg-peaks from an image? (I think yes)
2) If we have that, we could plot intensity around the ring for a Bragg reflection and see how it varies with all calibration parameters.
3) but this version could work out of the box as it is, but we need a bit more testing I think.